Marlin/src/module/motion.cpp

```diff
@@ -1049,6 +1049,17 @@ FORCE_INLINE void segment_idle(millis_t &next_idle_ms) {
 
 #endif // DUAL_X_CARRIAGE
 
+void plan_line_to_z(const float z, const feedRate_t fr) {
+  /// save default value
+  feedRate_t dfr = feedrate_mm_s;
+  destination = current_position;
+  destination.z += z;
```

Comment:
Since this adds the given value to the current Z position, this function should be called `prepare_z_raise`.